src/components/tree/Download.tsx

```diff
@@ -0,0 +1,48 @@
+import { FC, useLayoutEffect, useMemo, useState } from "react";
+
+import DownloadIcon from "../vectors/Download";
+import { TreeResult } from "../../types/result";
+
+type Props = {
+ treeSections: TreeResult
+}
+
+const Download: FC<Props> = ({ treeSections }) => {
+ const [uri, setUri] = useState<string>("")
```

Comment:
Could you create another state to follow the progress? You can use `const [status, setStatus] = useState<"idle" | "loading" | "success" | "error">("idle")`

So that you do something like:
```
setStatus("loading");
try {
  const csv = treeToCsv(treeSections);
  setStatus("success");
} catch (error) {
  setStatus("error");
  console.error(error); // Optional, I don't really know how the errors are being handled currently, you could do here what you are used to do to handle.
}
```
And then in the html you could show a feedback to the user, such as "Loading...", "There was an error, please try again...", "Your CSV has been downloaded...", or something like that.